app/services/policies/policies_list_service.rb

```diff
@@ -61,4 +56,21 @@ def to_h
       @sets.transform_values(&:to_a).as_json
     end
   end
+
+  HTTP_ERRORS = [HTTPClient::BadResponseError, HTTPClient::TimeoutError, HTTPClient::ConnectTimeoutError, HTTPClient::SendTimeoutError, HTTPClient::ReceiveTimeoutError, SocketError].freeze
+  private_constant :HTTP_ERRORS
+
+  def self.fetch_policies_from_apicast
+    return unless (apicast_registry_url = ThreeScale.config.sandbox_proxy.apicast_registry_url)
```

Comment:
Oh it is here where you should raise error and catch it in the method `self::call`

The feature is:

- if this is not set or there is a server error, then policies should not be displayed, i.e all policies (builtin and custom ones too)

- if it is set and works, display the policies.

Returning `nil` here does not carry this idea.

Reply:
Oh I see!